packages/jest-snapshot/src/utils.ts

```diff
@@ -178,13 +178,38 @@ export const saveSnapshotFile = (
   );
 };
 
+const deepMergeArray = (target: Array<any>, source: Array<any>) => {
+  // Clone target
+  const mergedOutput = target.slice();
+
+  source.forEach((sourceElement, index) => {
+    const targetElement = mergedOutput[index];
+    const targetElementType = typeof targetElement;
+
+    if (Array.isArray(target[index])) {
+      // Target is an array
+      mergedOutput[index] = deepMergeArray(target[index], sourceElement);
+    } else if (targetElementType === 'object') {
```

Comment:
`deepMerge` actually uses `isObject` here, probably because of `null` being an `object`

Reply:
Should this comment be addressed?

Reply:
Ah - I missed this. Updated.